snowflake/cmd/policyRecommendation.go

```diff
@@ -0,0 +1,285 @@
+TABLE(%s(
+  '%s',
+  '%s',
+  %d,
+  '%s',
+  pf_with_index.appliedTo,
+  pf_with_index.ingress,
+  pf_with_index.egress
+) over (partition by pf_with_index.appliedTo, pf_with_index.row_index)) as r
+`, functionName, jobType, recommendationID, isolationMethod, nsAllowList)
+
+	return query, nil
+}
+
+// policyRecommendationCmd represents the policy-recommendation command
+var policyRecommendationCmd = &cobra.Command{
+	Use:   "policy-recommendation",
+	Short: "Run the policy recommendation UDF in Snowflake",
+	Long: `This command runs the policy recommendation UDF in Snowflake. 
+You need to bring your own Snowflake account and created the policy
+recommendation UDF using the create-udfs command first.
+
+Run policy recommendation with default configuration on database ANTREA_C9JR8KUKUIV4R72S:
+"theia-sf policy-recommendation --database-name ANTREA_C9JR8KUKUIV4R72S"
+
+The "policy-recommendation" command requires a Snowflake warehouse to run policy
+recommendation UDFs in Snowflake. By default, it will create a temporary one.
+You can also bring your own by using the "--warehouse-name" parameter.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		jobType, _ := cmd.Flags().GetString("type")
+		if jobType != "initial" && jobType != "subsequent" {
+			return fmt.Errorf("invalid --type argument")
+		}
+		limit, _ := cmd.Flags().GetUint("limit")
+		isolationMethod, _ := cmd.Flags().GetInt("isolationMethod")
+		if isolationMethod < 1 && isolationMethod > 3 {
+			return fmt.Errorf("invalid -isolationMethod argument")
+		}
+		start, _ := cmd.Flags().GetString("start")
+		end, _ := cmd.Flags().GetString("end")
+		startTs, _ := cmd.Flags().GetString("start-ts")
+		endTs, _ := cmd.Flags().GetString("end-ts")
+		nsAllowList, _ := cmd.Flags().GetString("ns-allow")
+		labelIgnoreList, _ := cmd.Flags().GetString("label-ignore")
+		clusterUUID, _ := cmd.Flags().GetString("cluster-uuid")
+		databaseName, _ := cmd.Flags().GetString("database-name")
+		warehouseName, _ := cmd.Flags().GetString("warehouse-name")
+		functionVersion, _ := cmd.Flags().GetString("udf-version")
+		waitTimeout, _ := cmd.Flags().GetString("wait-timeout")
+		waitDuration, err := time.ParseDuration(waitTimeout)
+		if err != nil {
+			return fmt.Errorf("invalid --wait-timeout argument, err when parsing it as a duration: %v", err)
+		}
+		verbose := verbosity >= 2
+		query, err := buildPolicyRecommendationUdfQuery(jobType, limit, isolationMethod, start, end, startTs, endTs, nsAllowList, labelIgnoreList, clusterUUID, databaseName, functionVersion)
+		if err != nil {
+			return err
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), waitDuration)
+		defer cancel()
+		// stackName, stateBackendURL, secretsProviderURL, region, workdir are not provided here
+		// because we only uses snowflake client in this command.
+		mgr := infra.NewManager(logger, "", "", "", "", warehouseName, "", verbose)
+		rows, err := mgr.RunUdf(ctx, query, databaseName)
```

Comment:
I really think you should find a way to avoid using the infra manager here. Does `RunUdf` really need to be a method of the manager? You could use the Snowflake client directly.

Reply:
Agree, I was thinking reuseing `RunUdf` function in the future for other udf commands. Moved it under `infra/udfs` package now.